### PR TITLE
Text box implementation

### PR DESCRIPTION
**Pull request by fortwoone**
This PR adds a text box implementation that allows for character names to be displayed alongside text. The player cannot move while dialogue is displayed on-screen. 

(Debug features : press T to make an example piece of text appear, and press Space to advance the text or skip the animation.)

https://cdn.discordapp.com/attachments/829984222236114994/1257270292846350398/pydew_blahblah.mp4?ex=6683cbb4&is=66827a34&hm=4875e3a3457fd3b3e0c85d2ebe774f38038f647dfa0f526ac67bbb4b75a581c1&

### CODE REVIEW

**Comment by fortwoone** [comment]
Note that the text speed might be changed later (could be an additional setting).

File: main.py [hunk 1]
```diff
@@ -43,6 +45,9 @@ def __init__(self):
 
         # game setup
         self.running = True
+        self.level = Level(self, self.tmx_maps, self.character_frames, self.level_frames, self.overlay_frames, self.font,
```
Comment (Manunz15) [inline main.py:48]: Level must have the switch parameter

File: src/level.py [hunk 2]
```diff
@@ -24,8 +24,9 @@
 
 class Level:
 
-    def __init__(self, tmx_maps, character_frames, level_frames, overlay_frames, font, sounds, switch):
+    def __init__(self, game, tmx_maps: MapDict, character_frames, level_frames, overlay_frames, font, sounds, switch):
```
Comment (Manunz15) [inline src/level.py:27]: It seems like you added the 'game' parameter but didn't pass it in main.py

**Review by Manunz15 — COMMENTED** [review]
There is a bug when calling Level in main.py

File: src/level.py (hunk 2 above)
Comment (fortwoone) [reply src/level.py:27]: I definitely did, it must have disappeared when I resolved the merge conflicts. Going to change that

**Review by Manunz15 — APPROVED** [review]
Everything works fine

**Pull request merged**